tests/slsa_analyzer/checks/test_registry.py

```diff
@@ -200,3 +255,112 @@ def test_validate_check_id_format(self) -> None:
 
         assert all(Registry._validate_check_id_format(check_id) for check_id in valid_ids)
         assert all(not Registry._validate_check_id_format(check_id) for check_id in invalid_ids)
+
+
+@pytest.mark.parametrize(
+    ("ex_pats", "in_pats", "final_checks"),
+    [
+        (
+            [],
+            ["*"],
+            ["mcn_a_1", "mcn_b_1", "mcn_c_1", "mcn_d_1", "mcn_e_1", "mcn_f_1", "mcn_g_1", "mcn_h_1", "mcn_i_1"],
+        ),
+        (
+            [],
+            ["*", "*"],
+            ["mcn_a_1", "mcn_b_1", "mcn_c_1", "mcn_d_1", "mcn_e_1", "mcn_f_1", "mcn_g_1", "mcn_h_1", "mcn_i_1"],
+        ),
+        (
+            [],
+            ["mcn_?_1"],
+            ["mcn_a_1", "mcn_b_1", "mcn_c_1", "mcn_d_1", "mcn_e_1", "mcn_f_1", "mcn_g_1", "mcn_h_1", "mcn_i_1"],
+        ),
+        (
+            [],
+            ["mcn_[cf]_1"],
+            ["mcn_c_1", "mcn_f_1", "mcn_d_1", "mcn_e_1"],
+        ),
+        ([], [], []),
+        (["*"], [], []),
+        (["*"], ["*"], []),
+        (["*", "*", "*"], ["*", "*"], []),
+        (
+            [],
+            ["mcn_a_1", "mcn_b_1", "mcn_c_1", "mcn_d_1", "mcn_e_1", "mcn_f_1", "mcn_g_1", "mcn_h_1", "mcn_i_1"],
+            ["mcn_a_1", "mcn_b_1", "mcn_c_1", "mcn_d_1", "mcn_e_1", "mcn_f_1", "mcn_g_1", "mcn_h_1", "mcn_i_1"],
+        ),
+        (["mcn_c_1"], ["*"], ["mcn_d_1", "mcn_e_1", "mcn_g_1", "mcn_h_1", "mcn_i_1"]),
+        ([], ["mcn_c_1"], ["mcn_c_1", "mcn_d_1", "mcn_e_1"]),
+        (["mcn_d_1"], ["mcn_c_1"], ["mcn_e_1"]),
+        (["mcn_*"], ["*"], []),
+    ],
+)
+def test_get_final_checks(
+    check_registry: Registry, ex_pats: list[str], in_pats: list[str], final_checks: list[str]
+) -> None:
+    """This method tests the get_final_checks method."""
+    assert check_registry.get_final_checks(ex_pats=ex_pats, in_pats=in_pats).sort() == final_checks.sort()
```

Comment:
This is incorrect as `check_registry.get_final_checks(ex_pats=ex_pats, in_pats=in_pats).sort()` and `final_checks.sort()` would always return None.

Reply:
Fixed in https://github.com/oracle/macaron/pull/254/commits/0be70cbcb7eba77a441802b447da8f4361050f32